themes/default/content/blog/repairing-state-with-pulumi-refresh/index.md

```diff
@@ -0,0 +1,98 @@
+
+a. Started “pulumi up”.
+b. About to create `aws_native:s3:Bucket::my-bucket`.
+c. Created `aws_native:s3:Bucket::my-bucket` with ID `my-bucket-6e3d099`.
+d. Finished “pulumi up”.
+
+If the Pulumi CLI process is terminated between steps (b) and (c), then Pulumi does not know if
+“my-bucket” exists or not. Unlike (1), Pulumi is aware that it does not know if the resource was created. This will be displayed as a pending CREATE operation. Depending on what type of operation was interrupted, you can see pending CREATE, UPDATE, DELETE, READ and IMPORT operations.
+
+![Pending Operations Warning](pending-ops-warning.png)
+
+## The Solution
+
+As of [#10394](https://github.com/pulumi/pulumi/pull/10394), the solution to both problems is [pulumi refresh](https://www.pulumi.com/docs/reference/cli/pulumi_refresh/). The `pulumi refresh` command modifies Pulumi&rsquo;s state so it matches the state of the underlying providers. Most discrepancies can be resolved automatically by checking Pulumi&rsquo;s state against the underlying resource provider. For our aws bucket example, Pulumi would ask the aws-native provider if bucket `my-bucket-6e3d099` and what tags it has.
+
+This works for all resources that Pulumi knows about. This set includes Pulumi managed resources that were changed manually, or for pending UPDATE, DELETE, READ and IMPORT operations. Pulumi does not know the resource ID for pending CREATE operations (it might not exist), so it can&rsquo;t resolve everything for you. It needs a little bit of help. For each pending CREATE, Pulumi will ask you what to do:
+
+![Pulumi refresh asking for input](refresh-ask.png)
+
+If the resource was created successfully, you can select import, give Pulumi the ID of the
+created resource, and Pulumi will bring that resource under management.
+
+If no resource was created, you can tell Pulumi to clear the pending CREATE. It will recreate the resource for you the next time you run `pulumi up`.
+
+You can learn more by going to the [`pulumi refresh` documentation](https://www.pulumi.com/docs/reference/cli/pulumi_refresh/) or by running `pulumi refresh --help`.
+
+### Command Details
+
+There are some new flags you can pass to modify the new behavior:
+
+- `--clear-pending-creates` will remove any pending CREATEs from the state.
+- `--skip-pending-creates` will skip all pending CREATEs. This is the default behavior when running the CLI outside of interactive more.
+- `--import-pending-creates` gives Pulumi a list of URN ID pairs. This is equivalent to selecting the `import` option when running in interactive mode, but only for the URNs listed.
+
+These flags can be combined:
+
+```sh
+    pulumi refresh --import-pending-creates="aws_native:s3:Bucket::my-bucket my-bucket-6e3d099" --clear-pending-creates
```

Comment:
Little thing to consider, just so one doesn't have to scroll to see both options:

```suggestion
pulumi refresh \
    --import-pending-creates="aws_native:s3:Bucket::my-bucket my-bucket-6e3d099" \
    --clear-pending-creates
```